hotshot-state-prover/src/service.rs

```diff
@@ -315,7 +316,31 @@ pub async fn submit_state_and_proof(
     // prepare the input the contract call and the tx itself
     let proof: ParsedPlonkProof = proof.into();
     let new_state: ParsedLightClientState = public_input.into();
-    let tx = contract.new_finalized_state(new_state.into(), proof.into());
+
+    // frugal gas price: set to SafeGasPrice based on live price oracle
+    // safe to be included with low priority position in a block
+    let gas_info = reqwest::get("https://api.etherscan.io/api?module=gastracker&action=gasoracle")
+        .await?
+        .json::<serde_json::Value>()
+        .await?;
```

Comment:
I think we should handle the case where etherscan we get an error when calling the etherscan API.

Reply:
make sense, changed in [f0cf7aa](https://github.com/EspressoSystems/espresso-sequencer/pull/2148/commits/f0cf7aa08b69693e3fdd11d5bc17f4b8dbf7be26)

now, only set the price explicitly when etherscan successfully return; else simply ignore and create tx with default gas (whatever `ethers` decide to do)

Reply:
allow for falling back the default when Etherscan changes their API: [c835231](https://github.com/EspressoSystems/espresso-sequencer/pull/2148/commits/c835231830979279594e789707cf0a04d3239c59)